devices/debian.py

```diff
@@ -208,7 +208,12 @@ def run_cleanup_cmd(self):
     def sudo_sendline(self, s):
         if self.username != "root":
             s = "sudo " + s
-        return super(DebianBox, self).sendline(s)
+        super(DebianBox, self).sendline(s)
+        idx = self.expect(["password for .*:"] + self.prompt,timeout=20)
```

Comment:
So this is super tricky actually. You will need to avoid eating the output somehow...

I think you will have to run a test command to see if sudo prompts for a password
```
self.sendline(sudo true)
if 0 == expect(password for [^:]*:, prompt):
    will_prompt_for_password = True
else
    will_prompt_for_password = False

self.sendline(sudo + cmd)
if will_prompt_for_password:
    self.expect("password for .*:")
    self.sendline(self.password)
```

Reply:
Yes it expects for password at times. I have tried it, that is the reason I added the password expect prompt here